### PR TITLE
feat: Add trial metrics in wall time 

### PR DESCRIPTION
**Pull request by gt2345**
## Description

Partial task for ticket [WEB-660](https://determinedai.atlassian.net/browse/WEB-660)
Add metrics in wall time from backend to `/trials/compare` API.
As discussed with AMLE, we can use the metrics reported time as an additional x-axis for metrics.

## Test Plan

Ping `/trials/compare` with `x_axis=X_AXIS_TIME`, then metrics should have a field "time", and it is the metrics in time. 

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-660]: https://determinedai.atlassian.net/browse/WEB-660?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_trials.go [hunk 1]
```diff
@@ -649,11 +649,33 @@ func (a *apiServer) appendToMetrics(metrics []*apiv1.SummarizedMetric, m *apiv1.
 	return metrics
 }
 
+func timeFromFloat64(ts float64) time.Time {
+	secs := int64(ts)
+	nsecs := int64((ts - float64(secs)) * 1e9)
+	return time.Unix(secs, nsecs)
+}
```
Comment (gt2345) [inline master/internal/api_trials.go:656]: Do we have a dedicated place for utility functions like this?

**Comment by mapmeld** [comment]
ok I tested out the API. I see that this returns 

```
data: [
  {
    batches: 390,
    value: 4.22135066986084
  },
  ... { more metrics with batches/value } ...
],
time: [
  {
    time: "2023-01-05T15:06:19.719493865Z",
    value: 4.22135066986084
  },
  ... { more metrics with time/value } ...
],
```

I'm concerned about giving the text timestamp, and doubling the response size by including data twice via batches and time 
Maybe we should discuss it on Slack, but I would expect:
- time returned as Unix time (due to JavaScript limits on the Number type, this likely returns as a string)
- one API call should return only batches or only time

**Comment by gt2345** [comment]
@mapmeld , I changed the `/trials/compare` API to accept new parameter `x_axis=X_AXIS_TIME`.
About the timestamp, I was trying to follow how the protoc was set up in our code base, also I think it might be easier to just convert time in front end, because that where we can find the timezone info.

**Review by mapmeld — APPROVED** [review]
✅  thanks for the changes!

**Pull request merged**